src/handler/archive.ts

```diff
@@ -0,0 +1,70 @@
+import 'dotenv/config';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import logger from '../util/logger';
+import { validateArchiveErrors } from '../validators/archive';
+import { addHttpHeaders } from '../util/httpHeaders';
+import { archiveRecord, getBySystemNumberAndCreatedTimestamp } from '../services/database';
+import { Status } from '../util/enum';
+import { getUserDetails } from '../services/user';
+import { formatTechRecord } from '../util/formatTechRecord';
+import { ArchiveRecord, ArchiveRecordRequestBody } from '../models/archive';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Archive end point called');
+
+  const archiveErrors = validateArchiveErrors(event);
+
+  if (archiveErrors) {
+    return addHttpHeaders(archiveErrors);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const body: ArchiveRecordRequestBody = await JSON.parse(event.body as string);
+
+  if (!body.reasonForArchiving) {
```

Comment:
It could be however it would require the body to be parsed twice in order to keep the signature of the validate function to kept clean and in-line with the other validators.